language/stdlib/modules/LibraAccount.move

```diff
@@ -332,7 +343,10 @@ module LibraAccount {
             );
             assert(can_withdraw, EWITHDRAWAL_EXCEEDS_LIMITS);
         };
-        Libra::withdraw(&mut balance.coin, amount)
+        let coin = &mut balance.coin;
+        // Abort if this withdrawal would make the `payer`'s balance go negative
+        assert(Libra::value(coin) >= amount, EINSUFFICIENT_BALANCE);
```

Comment:
This check happens in the transaction prologue. If the sender of the tx doesn't have sufficient gas, it will be discarded before the tx script actually gets executed.

Reply:
ok great, sounds good.